packages/@vuepress/theme-default/components/NavLink.vue

```diff
@@ -39,13 +39,28 @@ export default {
         return Object.keys(this.$site.locales).some(rootLink => rootLink === this.link)
       }
       return this.link === '/'
+    },
+
+    target () {
+      return isMailto(this.link) || isTel(this.link) ? null : this.item.target
+      || (isExternal(this.link) ? '_blank' : '')
+    },
+
+    isTargetBlank () {
+      return this.target === '_blank'
+    },
+
+    rel () {
+      return isMailto(this.link) || isTel(this.link) ? null : this.item.rel
+      || (this.isTargetBlank ? 'noopener noreferrer' : '')
+    },
+
+    isInternal () {
+      return !isExternal(this.link) && !this.isTargetBlank
     }
```

Comment:
With many trinomial expressions. if not extra parentheses, the code will not be able to read intuitively.

So I personally have following codes suggested:


```js
  isNonHttpURI () {
    return isMailto(this.link) || isTel(this.link)
  },

  isInternal () {
    return !isExternal(this.link) && !this.isBlankTarget
  },

  isBlankTarget () {
    return this.target === '_blank'
  },

  target () {
    if (this.isNonHttpURI()) {
      return null
    }

    if (this.item.target) {
      return this.item.target
    }

    return isExternal(this.link)
      ? '_blank' : ''
  },

  rel () {
    if (this.isNonHttpURI()) {
      return null
    }

    if (this.item.rel) {
      return this.item.target
    }

    return this.isBlankTarget
      ? 'noopener noreferrer'
      : ''
  }
```